src/modules/SearchPage/index.tsx

```diff
@@ -1,85 +1,70 @@
-import { SearchBar, Tabs, Cards } from './components';
+import { AvailableTabs, INPUT_VALUE_KEY } from 'appConstants';
+import { sortMove, sortPokemon, sortType } from 'utils';
+import { useAppContext } from 'contexts';
+import { Loader } from 'components';
 import { Layout } from 'modules';
 import * as S from './styled';
-import {
-  fetchPokemonByParameter,
-  fetchMoveByParameter,
-  fetchTypeByParameter,
-  fetchPokemonDetails,
-  fetchPokemonBase,
-  fetchMoveDetails,
-  fetchTypeDetails,
-  fetchTypeBase,
-  fetchMoveBase,
-} from 'utils';
 
 type Props = {
   componentName: string;
   location: string;
 };
 
 export const SearchPage = ({ componentName, location }: Props) => {
-  const [lengths, setLengths] = useState<Lengths>({ moves: 0, pokemons: 0, types: 0 });
-  const [activeTab, setActiveTab] = useState<AvailableTabs>(AvailableTabs.pokemons);
-  const [pokemons, setPokemons] = useState<Array<PokemonDetails>>([]);
-  const [moves, setMoves] = useState<Array<PokemonMoveDetails>>([]);
-  const [types, setTypes] = useState<Array<PokemonTypeDetails>>([]);
   const [inputValue, setInputValue] = useState('');
-  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    const inputValue = window.localStorage.getItem(INPUT_VALUE_KEY);
-    if (inputValue && inputValue.trim().length > 0) {
-      setInputValue(inputValue);
-      fetchAndSetSpecificData(inputValue).then(() => setIsLoading(false));
-    } else fetchAndSetAllData().then(() => setIsLoading(false));
+    const value = window.localStorage.getItem(INPUT_VALUE_KEY);
+    value && setInputValue(value);
   }, []);
 
-  const fetchAndSetSpecificData = async (value: string) => {
-    const type = await fetchTypeByParameter(`${API.TYPE}/${value}`);
-    const move = await fetchMoveByParameter(`${API.MOVE}/${value}`);
-    const pokemon = await fetchPokemonByParameter(`${API.POKEMON}/${value}`);
-    setPokemons(pokemon ? [pokemon] : []);
-    setTypes(type ? [type] : []);
-    setMoves(move ? [move] : []);
-    setLengths({
-      pokemons: pokemon ? 1 : 0,
-      moves: move ? 1 : 0,
-      types: type ? 1 : 0,
-    });
-  };
-
-  const fetchAndSetAllData = async () => {
-    const pokemons = (await fetchPokemonBase(`${API.POKEMON}${API.POKEMON_LIMIT}`)) || [];
-    const pokemonsDetailed = (await fetchPokemonDetails(pokemons)) || [];
-    const moves = (await fetchMoveBase(`${API.MOVE}${API.MOVE_LIMIT}`)) || [];
-    const movesDetailed = (await fetchMoveDetails(moves)) || [];
-    const types = (await fetchTypeBase(`${API.TYPE}${API.TYPE_LIMIT}`)) || [];
-    const typesDetailed = (await fetchTypeDetails(types)) || [];
-    setPokemons(pokemonsDetailed);
-    setTypes(typesDetailed);
-    setMoves(movesDetailed);
-    setLengths({
-      pokemons: pokemonsDetailed.length,
-      moves: movesDetailed.length,
-      types: typesDetailed.length,
-    });
-  };
+  const { allDataResults, resultsAmount, isLoading, activeTab } = useAppContext().state;
```

Comment:
try to not to use such unclear syntax **useAppContext().state;**

Reply:
changed